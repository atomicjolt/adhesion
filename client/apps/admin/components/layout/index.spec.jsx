import React from 'react';
import { shallow } from 'enzyme';
import { Index } from './index';

describe('layout index', () => {
  let result;

  let getSites = false;
  let getApplications = false;

  let props;

  const text = 'hello';
  const children = <h1>{text}</h1>;

  beforeEach(() => {
    props = {
      children,
      getApplications: () => { getApplications = true; },
      getSites: () => { getSites = true; },
      location: {
        pathname: '/'
      },
    };
    result = shallow(<Index {...props} />);
  });

  it('renders the index', () => {
    expect(result).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('Loads sites', () => {
    result.instance().componentDidMount();
    expect(getSites).toBe(true);
  });

  it('Loads applications', () => {
    expect(getApplications).toBe(true);
  });
});
