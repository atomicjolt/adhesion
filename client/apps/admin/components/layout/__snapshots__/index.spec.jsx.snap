// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout index matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index
    getApplications={[Function]}
    getSites={[Function]}
    location={
        Object {
            "pathname": "/",
          }
    }
>
    <h1>
        hello
    </h1>
</Index>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        hello
</h1>,
      "className": "app-index",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "hello",
      },
      "ref": null,
      "rendered": "hello",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          hello
</h1>,
        "className": "app-index",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
